src/olympia/devhub/tests/test_views_submit.py

```diff
@@ -84,34 +86,72 @@ def get_version(self):
 
 
 class TestAddonSubmitAgreement(TestSubmitBase):
-    def test_step1_submit(self):
+    def test_submit_agreement_page_links(self):
         self.user.update(read_dev_agreement=None)
         response = self.client.get(reverse('devhub.submit.agreement'))
         assert response.status_code == 200
         doc = pq(response.content)
-        links = doc('#agreement-container a')
+        links = doc('.agreement-links a')
         assert links
         for ln in links:
             href = ln.attrib['href']
-            assert not href.startswith('%'), (
+            assert href.startswith(('https://', '/', 'mailto:')), (
                 "Looks like link %r to %r is still a placeholder" %
                 (href, ln.text))
 
-    def test_read_dev_agreement_set(self):
+    def test_set_read_dev_agreement(self):
         """Store current date when the user agrees with the user agreement."""
         self.user.update(read_dev_agreement=None)
 
-        response = self.client.post(reverse('devhub.submit.agreement'),
-                                    follow=True)
-        user = response.context['user']
-        self.assertCloseToNow(user.read_dev_agreement)
+        response = self.client.post(reverse('devhub.submit.agreement'))
+        assert response.status_code == 302
+        self.user.reload()
+        self.assertCloseToNow(self.user.read_dev_agreement)
 
     def test_read_dev_agreement_skip(self):
         # The current user fixture has already read the agreement so we skip
         response = self.client.get(reverse('devhub.submit.agreement'))
         self.assert3xx(response, reverse('devhub.submit.distribution'))
 
 
+@override_switch('post-review', active=True)
```

Comment:
if it's obvious, can you add an `@override_switch('post-review', active=False)` to the test classes that would fail after the waffle is removed?